packages/alfa-dom/src/node/query/element-descendants.ts

```diff
@@ -0,0 +1,23 @@
+import { Cache } from "@siteimprove/alfa-cache";
+import { Sequence } from "@siteimprove/alfa-sequence";
+import { Node } from "../../node";
+import { Element } from "../element";
+
+const _elementDescendantsCache = Cache.empty<Node, Array<Sequence<Element>>>();
+
+/**
+ * @public
+ */
+export function getElementDescendants(
+  node: Node,
+  options: Node.Traversal = Node.Traversal.empty
+): Sequence<Element> {
+  const optionsMap = _elementDescendantsCache.get(node, () => []);
+  if (optionsMap[options.value] === undefined) {
+    optionsMap[options.value] = node
+      .descendants(options)
+      .filter(Element.isElement);
```

Comment:
**Issue:** Does that actually update the array inside the cache 🤔

Reply:
> It seems to yes when I step through the code, I guess because it's just a reference to the physical array or something?

👍 

> Not sure however if it's the best approach. Maybe it's better to explicitly have a 4 separate caches for each of the possible enum values?

No. There are actually 8 enum values (this is a bits array, they can be combined: https://github.com/Siteimprove/alfa/blob/main/packages/alfa-dom/src/node.ts#L319). The `.value` helper is precisely here so that we abstract the flag implementation away and get a guaranteed different integer for each flag combination. See [alfa-flags](https://github.com/Siteimprove/alfa/blob/main/packages/alfa-flags/src/flags.ts) for the dirty details.